### DISCUSSION TITLE
ED060TC1 display with 35 pins - an Amazon Kindle Voyage screen

### DISCUSSION THREAD

**Discussion by vanarebane** [vroland/epdiy#293] [Q&A]
Hi, I got my hands on a Amazon Kindle Voyage with damaged motherboard. The screen seems fine and I was researching if I can drive that screen with some DIY hardware.

I was really full of hope when I wound out the screen is 1072×1448 pixel 6" E-ink Carta in the ED060K## lineup. The Kindle screen has model [ED060TC1](https://guide-images.cdn.ifixit.com/igi/HZ5unQnUkVUMQsoJ.full) and looking at the capacitors on the connector side I though it initially was [ED060KD1](https://www.panelook.com/ED060KD1__6.0__overview_38250.html) (by comparing pictures from online). But I found out that the connector is 35 pin and much smaller pitch, 0.3mm, in two rows, 10.5mm wide. [Similar to this](https://ee.farnell.com/molex/15032-0235/cable-assy-fpc-35pos-51mm-brn/dp/4036946?st=connectors%200.3mm%2035pin%20flex)

Doing a little research, the connector seems to be [Molex 5035663502](https://www.molex.com/en-us/products/part-detail/5035663502)

And doing a little more comparison, The Kindle ED060TC1 connector goes to the left, while ED060KD1 goes to the right.

Can anyone help, advice or datasheet wise, how to detect pinout for the ED060TC1 display?
I'd like to build an adapter for converting it to the 34 pin 0.5 mm pitch so it can be connected to Epdiy.
I have some reverse-engineering skills and PCB design skills, but not a good understanding how to reverse-engineer a 0.3mm pitch display connector.

**Comment by vroland** [comment]
Hi, I'm always in favor of recycling and hacking hardware! :+1: Hm, the easiest would be to have a datasheet. Then designing a simple adapter should become easy. 
Without it, reversing the connections can become tricky... If you want to spend the time, you could make a breakout board for the connector and get an FPC to connect to the eReader and just break out the connections. Then you could hope that the display power still turns on and you can figure out the high voltage power lines. For the data lines you'd need a logic analyzer or oscilloscope, but the risk of damage is much lower once you stay clear of the power lines. 
If the reader doesn't give a supply voltage anymore then you could try to separate the power lines by trace width or something, but the polarity will be hard to figure out. Maybe you can speculate on the order being the same as other displays :sweat_smile:

**Comment by martinberlin** [comment]
I also like hacking and seeing new displays refresh!
Since this [Molex connector](https://jlcpcb.com/partdetail/Molex-5035663302/C585270) is also in LCSC/ JLCPCB I can help if you can find the datasheet. I think it will be the best to ask for an adapter using PCB Assembly since to solder the 0.3 mm pitch can be  quite a daunting job (At least for my skills)
So just tell if I can give a hand here and will find some time to make an adapter.

**Comment by vanarebane** [comment]
@vroland This is going to be long winded project, to reverse it. What I was told is a bad motherboard turned out to be a cracked screen instead. I opened it up for better access and while on surface a e-ink looks fine, underneath can be cracked. So I have less motivation to reverse-engineer it myself. First I need to acquire a EPDIY also. Perhaps I can remove the flex cable from the screen now instead and follow traces physically. @martinberlin thanks for the help, I myself use solder paste and stencils, so 0.3mm is doable. I'm planning to do a PCB order soon, so I'll add the adapter for a breakoff to some existing design.

Not sure if ED060TC1 might be worth reverse engineering further. I put the screen into my high resolution photon scanner.
This is my initial guesses [ED060TC1 reverse engineering part1.pdf](https://github.com/vroland/epdiy/files/13461442/ED060TC1.reverse.engineering.part1.pdf), here is the legend
* black - probably GND
* orange - wider traces, might be power
* red - direct signals, thin lines
* green - some signals and SMD mixed

It's interesting for me to analyse traces and how they are designed. If someone has ED060KC1 or ED060KD1, having a scan of those screen PCB and it would help to name the pins.

**Comment by vroland** [comment]
Hm, if the screen is the faulty part though, you should get valid signals from the e-reader though, right? It's probably still hard to probe the traces so that may require making a breakout, but it's a place to start. Though at this point I agree, you could just order another display instead :D

**Comment by zephray** [comment]
Something might be helpful, the ED060XC8 uses the same 35pin 0.3mm 2 rows connector, with the following pinout:
```
1. VGH
2. VSS
3. VSS
4. F_CS
5. VPOS
6. F_MOSI
7. NC
8. F_SCLK
9. VGL
10. S_MISO
11. VSS
12. SPV
13. VNEG
14. CKV
15. NC
16. MODE
17. VCOM
18. D7
19. BORDER
20. D6
21. F_VDD
22. D5
23. VSS
24. D4
25. VDD
26. D3
27. VSS
28. D2
29. XLE
30. D1
31. XCL
32. D0
33. XOE
34. XSTL
35. VSS
```
It does look like it matches your pinout reasonably well, but your screen does have other stuff connected on the NC pins.

This pinout is what I reversed engineered on an ED060XC8 and verified to be working. If this is still something worthwhile maybe I can order a ED060TC1 to just plug it into my board and see what happens...

Note: my adapter to adapt from the 40pin 7.8" interface to this 35pin interface: https://github.com/zephray/NekoInk/tree/master/pcb/35p-adapter

**Comment by martinberlin** [comment]
Thanks a lot for the pinout @zephray 

> So I have less motivation to reverse-engineer it myself. First I need to acquire a EPDIY also. Perhaps I can remove the flex cable from the screen now instead and follow traces physically. 

@vanarebane is there still interest in getting this panel to work?

**Comment by vanarebane** [comment]
First off, @zephray thanks for the pinout. It seems to match the ED060TC1 indeed.

Between Pin 7 and 15 there is a 10K resistor, probably for some display detection for Amazon.
So the ED060TC1 pinout is as follows:
```
 1. VGH
 2. VSS
 3. VSS
 4. F_CS
 5. VPOS
 6. F_MOSI
 7. 10k resistor between pin 15
 8. F_SCLK
 9. VGL
10. S_MISO
11. VSS
12. SPV
13. VNEG
14. CKV
15. 10k resistor between pin 7
16. MODE
17. VCOM
18. D7
19. BORDER
20. D6
21. F_VDD
22. D5
23. VSS
24. D4
25. VDD
26. D3
27. VSS
28. D2
29. XLE
30. D1
31. XCL
32. D0
33. XOE
34. XSTL
35. VSS
``` 

@martinberlin, since my salvaged display turned out to be broken, I ordered ED060KC1 instead, as it made more sense to order a proven and supported e-ink instead of a unknown display.

@zephray, I'll let you decide if you wish to order a ED060TC1 and test it out. You could get it even with the Kindle front panel, so might be in some use to integrate it in some project. 

If not, just archive it so if someone has a Kindle Paperwhite with bad motherboard, or without any use, they now can salvage that screen. Or if the drivers are not different, make an adapter for the Kindle Paperwhite to drive a larger display, if the resolution is the same and the code does not change. Why? I can't come up with any valid reason other than bragging rights and for the pleasure of hacking :)

**Comment by tomek-szczesny** [comment]
> Between Pin 7 and 15 there is a 10K resistor, probably for some display detection for Amazon.

ES108 series required 10k resistor between two pins to terminate factory test mode. 
One pin is marked "XON", and the other is VDD.
I would check for shorts between either of these pins and VDD.

**Comment by martinberlin** [comment]
Converting this to discussions since it’s not a specific problem or issue with the epdiy component.
About this topic I was thinking the other day that it would be great to make a small epdiy v7 version with 2 default connectors that accepts HATs. So exposes 8 data pin plus signals and you can mount a HAT on top to support a different display.
Just an idea of course
